mmseg/core/optimizers/layer_decay_optimizer_constructor.py

```diff
@@ -146,3 +146,85 @@ def add_params(self, params, module):
                 }
             logger.info(f'Param groups = {json.dumps(to_display, indent=2)}')
         params.extend(parameter_groups.values())
+
+
+def get_num_layer_for_vit(var_name, num_max_layer):
+    """Get the layer id to set the different learning rates.
+
+    Args:
+        var_name (str): The key of the model.
+        num_max_layer (int): Maximum number of backbone layers.
+    Returns:
+        layer id (int): Returns the layer id of the key.
+    """
+
+    if var_name in ('backbone.cls_token', 'backbone.mask_token',
+                    'backbone.pos_embed'):
+        return 0
+    elif var_name.startswith('backbone.patch_embed'):
+        return 0
+    elif var_name.startswith('backbone.layers'):
+        layer_id = int(var_name.split('.')[2])
+        return layer_id + 1
+    else:
+        return num_max_layer - 1
+
+
+@OPTIMIZER_BUILDERS.register_module()
+class LayerDecayOptimizerConstructor(DefaultOptimizerConstructor):
```

Comment:
Can we merge `LayerDecayOptimizerConstructor` and `LearningRateDecayOptimizerConstructor` to one class?

Reply:
Yep.

Reply:
must be add a deprecated warning for the deleted one